tests/pytests/unit/modules/test_junos.py

```diff
@@ -0,0 +1,2852 @@
+            "hostname": "R1_re0",
+            "hostname_info": {"re0": "R1_re0", "re1": "R1_re01"},
+            "ifd_style": "CLASSIC",
+            "junos_info": {
+                "re0": {
+                    "object": {
+                        "build": None,
+                        "major": (16, 1),
+                        "minor": "20160413_0837_aamish",
+                        "type": "I",
+                    },
+                    "text": "16.1I20160413_0837_aamish",
+                },
+                "re1": {
+                    "object": {
+                        "build": None,
+                        "major": (16, 1),
+                        "minor": "20160413_0837_aamish",
+                        "type": "I",
+                    },
+                    "text": "16.1I20160413_0837_aamish",
+                },
+            },
+            "master": "RE0",
+            "model": "MX240",
+            "model_info": {"re0": "MX240", "re1": "MX240"},
+            "personality": "MX",
+            "re_info": {
+                "default": {
+                    "0": {
+                        "last_reboot_reason": "0x200:normal shutdown",
+                        "mastership_state": "master",
+                        "model": "RE-VMX",
+                        "status": "OK",
+                    },
+                    "1": {
+                        "last_reboot_reason": "0x200:normal shutdown",
+                        "mastership_state": "backup",
+                        "model": "RE-VMX",
+                        "status": "OK",
+                    },
+                    "default": {
+                        "last_reboot_reason": "0x200:normal shutdown",
+                        "mastership_state": "master",
+                        "model": "RE-VMX",
+                        "status": "OK",
+                    },
+                }
+            },
+            "re_master": {"default": "0"},
+            "serialnumber": "VMX4eaf",
+            "srx_cluster": None,
+            "switch_style": "BRIDGE_DOMAIN",
+            "vc_capable": False,
+            "vc_fabric": None,
+            "vc_master": None,
+            "vc_mode": None,
+            "version": "16.1I20160413_0837_aamish",
+            "version_RE0": "16.1I20160413_0837_aamish",
+            "version_RE1": "16.1I20160413_0837_aamish",
+            "version_info": {
+                "build": None,
+                "major": (16, 1),
+                "minor": "20160413_0837_aamish",
+                "type": "I",
+            },
+            "virtual": True,
+        }
+        ret["out"] = True
+        assert junos.facts_refresh() == ret
+
+
+def test_facts_refresh_exception():
+    with patch("jnpr.junos.device.Device.facts_refresh") as mock_facts_refresh:
+        mock_facts_refresh.side_effect = raise_exception
+        ret = dict()
+        ret["message"] = 'Execution failed due to "Test exception"'
+        ret["out"] = False
```

Comment:
```suggestion
        ret {
            "message": 'Execution failed due to "Test exception"',
            "out": False,
        }
```
What's up with these dictionaries getting constructed in chunks when there really isn't a need for?

Reply:
ported from unittests, don't blame the porter :) 
done

Reply:
Some cleanup when porting is always helpful.